firebase/ManageEvents.ts

```diff
@@ -49,18 +47,16 @@ export const getAllPastEvents = async () => {
     const eventsRef = collection(db, "events")
 
     // Create a query against the collection.
-    // This query retrieves events where the eventDate is less than the current date and time.
-    const q = query(eventsRef, where("date", "<", new Date()), orderBy("date", "desc"))
+    const nowIsoString = new Date().toISOString()
```

Comment:
Just a little detail - users in different time zones will have different current time returned by `new Date()`. Therefore, some events for such users may appear as still upcoming even though they are already done. This is just a problem that I have encountered in real life and unfortunately the only solution is to make these checks in the backend. This unfortunately cannot be done in Firebase.

Reply:
Interesting! I did not have that perspective, such a pain in the ass haha